lib/galaxy/tools/toolbox/lineages/factory.py

```diff
@@ -2,6 +2,16 @@
 from .stock import StockLineage
 
 
+def remove_version_from_guid( guid ):
+    """
+    Removes version from toolshed-derived tool_id(=guid).
+    """
+    if "/repos/" not in guid:
+        return None
+    tool_version = guid.split("/")[-1]
+    return guid.split(tool_version)[0]
```

Comment:
This may be unsafe if tool_version appears somewhere else in `guid`, I think the following is better:
```python
last_slash = guid.rfind('/')
return guid[:last_slash] # or guid[:last_slash + 1] if you need the trailing slash
```